pkg/prometheus/agent/operator.go

```diff
@@ -591,6 +603,19 @@ func statefulSetKeyToPrometheusAgentKey(key string) (bool, string) {
 	return true, matches[0][1] + "/" + matches[0][2]
 }
 
+func daemonSetKeyToPrometheusAgentKey(key string) (bool, string) {
```

Comment:
nit:

A comment explaining what this function does would help a ton, an example in the comment would be even better. I know `statefulSetKeyToPrometheusAgentKey` doesn't have a comment as well, but a comment there would also be useful